src/core/server.js

```diff
@@ -505,11 +513,13 @@ Server.prototype.connect = function() {
       self.emit('socket_error');
 
       if (self._remote.trace) {
-        log.info(self.getServerID(), 'onerror:', e.data || e);
+        log.info(`${self.getServerID()} onerror: ${e.toString()}`);
       }
 
-      if (Server.TLS_ERRORS.indexOf(e.message) !== -1) {
-        // Unrecoverable
+      if (Server.TLS_ERRORS.includes(e.message)) {
+        throw e;
+      }
+      if (e.message.includes('unexpected')) {
```

Comment:
Not sure if this is going to be caught in the browser https://github.com/websockets/ws/blob/7debd827d39925650bba5a270811ed82616f0eb1/lib/WebSocket.js#L693